compiler/ir/ir.tree/src/org/jetbrains/kotlin/ir/util/ConstantValueGenerator.kt

```diff
@@ -36,23 +36,32 @@ abstract class ConstantValueGenerator(
         startOffset: Int,
         endOffset: Int,
         constantValue: ConstantValue<*>,
-        varargElementType: KotlinType? = null
     ): IrExpression =
         // Assertion is safe here because annotation calls and class literals are not allowed in constant initializers
-        generateConstantOrAnnotationValueAsExpression(startOffset, endOffset, constantValue, null, varargElementType)!!
+        generateConstantOrAnnotationValueAsExpression(startOffset, endOffset, constantValue, null, null)!!
 
     /**
      * @return null if the constant value is an unresolved annotation or an unresolved class literal
      */
+    fun generateAnnotationValueAsExpression(
+        startOffset: Int,
+        endOffset: Int,
+        constantValue: ConstantValue<*>,
+        valueParameter: ValueParameterDescriptor,
+    ): IrExpression? =
+        generateConstantOrAnnotationValueAsExpression(
+            startOffset, endOffset, constantValue, valueParameter.type, valueParameter.varargElementType
+        )
+
     private fun generateConstantOrAnnotationValueAsExpression(
         startOffset: Int,
         endOffset: Int,
         constantValue: ConstantValue<*>,
-        realType: KotlinType?,
-        varargElementType: KotlinType? = null
+        inferredType: KotlinType?,
```

Comment:
Minor, but not sure if "inferred" is the perfect name... isn't it more of an "expected" type?